### ISSUE TITLE
Error in xgb.iter.update(bst$handle, dtrain, iteration - 1, obj) :    [03:26:03] amalgamation/../src/objective/multiclass_obj.cc:75: Check failed: label_error >= 0 && label_error < nclass SoftmaxMultiClassObj: label must be in [0, num_class), num_class=2 but found 2 in label.

### ISSUE THREAD

**Issue by itskdv** [dmlc/xgboost#3065]
[Xgboost .txt](https://github.com/dmlc/xgboost/files/1658413/Xgboost.txt)

Getting below errror , need help
Error in xgb.iter.update(bst$handle, dtrain, iteration - 1, obj) : 
  [03:26:03] amalgamation/../src/objective/multiclass_obj.cc:75: Check failed: label_error >= 0 && label_error < nclass SoftmaxMultiClassObj: label must be in [0, num_class), num_class=2 but found 2 in label.

## Environment info
Operating System: Win 10

Compiler:

Package used (python/R/jvm/C++): R

`xgboost` version used: Latest version 0.6-4

If installing from source, please provide 

1. The commit hash (`git rev-parse HEAD`)
2. Logs will be helpful (If logs are large, please upload as attachment).

If you are using jvm package, please 

1. add [jvm-packages] in the title to make it quickly be identified
2. the gcc version and distribution

If you are using python package, please provide

1. The python version and distribution
2. The command to install `xgboost` if you are not installing from source

If you are using R package, please provide

1. The R `sessionInfo()` 
install.packages("xgboost")
library(xgboost)
library(magrittr)
library(dplyr)
library(Matrix)
str(titanic.full)
2. The command to install `xgboost` if you are not installing from source

## Steps to reproduce

1.Run the txt file in R
2.
3.

## What have you tried?

1. Searched on Google but could not find solution
2.
3.

**Comment by khotilov** [comment]
You are assigning a factor to DMatrix label, which is implicitely converted to integer starting from 1. As the error message says, the "label must be in [0, num_class)". So, convert the factor to integer and subtract one.
